packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -1,96 +1,151 @@
-                        <h1 className={classes.productName}>
-                            <span>{data.name}</span>
-                        </h1>
-                        <p className={classes.productPrice}>
-                            <span>{data.price}</span>
-                        </p>
-                    </section>
-                    <section className={classes.imageCarousel}>
-                        <Carousel images={data.images} />
-                    </section>
-                    <section className={classes.actions}>
-                        <button className={classes.action}>
-                            <span>Add to Wishlist</span>
-                        </button>
-                    </section>
-                    <section className={classes.options}>
-                        <Options options={data.options} />
-                    </section>
-                    <section className={classes.quantity}>
-                        <h2 className={classes.quantityTitle}>
-                            <span>Quantity</span>
-                        </h2>
-                        <Quantity />
-                    </section>
-                    <section className={classes.cartActions}>
-                        <button className={classes.addToCart}>
-                            <span>Add to Cart</span>
-                        </button>
-                    </section>
-                    <section className={classes.description}>
-                        <h2 className={classes.descriptionTitle}>
-                            <span>Product Description</span>
-                        </h2>
-                        <RichText content={data.description} />
-                    </section>
-                    <section className={classes.details}>
-                        <h2 className={classes.detailsTitle}>
-                            <span>Details</span>
-                        </h2>
-                        <RichText content={data.additionalInfo} />
-                    </section>
-                    <section className={classes.related}>
-                        <h2 className={classes.relatedTitle}>
-                            <span>Similar Products</span>
-                        </h2>
-                    </section>
-                </article>
+                <Query
+                    query={productDetailQuery}
+                    variables={{ urlKey: getUrlKey() }}
+                >
+                    {({ loading, error, data }) => {
+                        if (error) return <div>Data Fetch Error</div>;
+                        if (loading) return <div>Fetching Data</div>;
+
+                        const product = data.productDetail.items[0];
+                        const { regularPrice } = product.price;
+
+                        return (
+                            <article className={classes.root}>
+                                <section className={classes.title}>
```

Comment:
Most of the keys we're doing lookups on with `classes` are missing in the proptypes definition for this component.

Reply:
CSS classes are too fast-changing to solidify as a type; they're just a dictionary of strings, so I changed the prop type to that.

Reply:
I think you should discuss this with @jimbo before deviating. There is a good reason these are documented the way they are. They're a public API

Reply:
> There is a good reason these are documented the way they are. They're a public API.

This. If you're writing CSS, you're targeting classnames. If those classnames are changing, your CSS needs changing. That's an interface.

> CSS classes are too fast-changing to solidify as a type

This really shouldn't be the case with CSS modules. Since the solution to a problem is never "add a classname to it", the classnames shouldn't really change unless a structure change necessitates it. As with any API, if you're changing it too frequently, you should probably be thinking about it more.

Reply:
Those are fair points; the `objectOf` prop type seems a bit useless as a consequence. Let's add all the classnames in a later PR; the missing `classes` prop types [predate this PR](https://github.com/magento-research/pwa-studio/blob/master/packages/venia-concept/src/RootComponents/Product/Product.js#L15-L17), and we don't want this one to get too large. I'll revert the change I made and get rid of the `objectOf` proptype.

Reply:
It's ok, I fixed it for you. 🙂